frontend/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -142,11 +142,13 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
     const { element, width, height, theme } = props
     const json = JSON.parse(element.json)
 
-    // Use either the Mapbox light or dark style based on Streamlit's theme
+    // If unset, use either the Mapbox light or dark style based on Streamlit's theme
     // For Mapbox styles, see https://docs.mapbox.com/api/maps/styles/#mapbox-styles
-    const hasLightBg = getLuminance(theme.colors.bgColor) > 0.5
-    const mapTheme = hasLightBg ? "light" : "dark"
-    json.mapStyle = `mapbox://styles/mapbox/${mapTheme}-v10`
+    if (typeof json.mapStyle === "undefined") {
```

Comment:
I think you can also just check here via `json.mapStyle === undefined` or even use this utils method:

https://github.com/streamlit/streamlit/blob/720a8497788754ec946dfeb6fb382caa3b92abc6/frontend/src/lib/utils.ts#L118

Reply:
Fixed on 247e918.